pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+)
+
+// MockClient mocks out an S3 bucket
+type MockClient struct {
```

Comment:
If we're actually writing to the filesystem when not just have the caller specify a local path for their testing and use the pathio code itself?